packages/deployments/contracts/contracts/nomad-xapps/contracts/relayer-fee-router/RelayerFeeRouter.sol

```diff
@@ -0,0 +1,190 @@
+
+  /**
+   * @notice Emitted when a fees claim has been initialized in this domain
+   * @param domain The domain where to claim the fees
+   * @param recipient The address of the relayer
+   * @param amount The amount of fees to claim
+   * @param transactionIds A group of transaction ids to claim for fee bumps
+   * @param remote Remote RelayerFeeRouter address
+   * @param message The message sent to the destination domain
+   */
+  event Send(uint32 domain, address recipient, uint256 amount, bytes32[] transactionIds, bytes32 remote, bytes message);
+
+  /**
+   * @notice Emitted when the a fees claim message has arrived to this domain
+   * @param originAndNonce Domain where the transfer originated and the unique identifier
+   * for the message from origin to destination, combined in a single field ((origin << 32) & nonce)
+   * @param origin Domain where the transfer originated
+   * @param recipient The address of the relayer
+   * @param amount The amount of fees to claim
+   * @param transactionIds A group of transaction ids to claim for fee bumps
+   */
+  event Receive(
+    uint64 indexed originAndNonce,
+    uint32 indexed origin,
+    address indexed recipient,
+    uint256 amount,
+    bytes32[] transactionIds
+  );
+
+  /**
+   * @notice Emitted when a new Connext address is set
+   * @param connext The new connext address
+   */
+  event SetConnext(address indexed connext);
+
+  // ======== Receive =======
+  receive() external payable {}
+
+  fallback() external payable {}
+
+  // ============ Modifiers ============
+
+  /**
+   * @notice Restricts the caller to the local bridge router
+   */
+  modifier onlyConnext() {
+    if (msg.sender != address(connext)) revert RelayerFeeRouter__onlyConnext_notConnext();
+    _;
+  }
+
+  // ======== Initializer ========
+
+  function initialize(address _xAppConnectionManager, address _connext) public initializer {
+    connext = IConnext(_connext);
+    __XAppConnectionClient_initialize(_xAppConnectionManager);
+
+    emit SetConnext(_connext);
+  }
+
+  /**
+   * @notice Sets the transaction manager.
+   * @dev Connext and relayer fee router store references to each other
+   * @param _connext The address of the Connext implementation
+   */
+  function setConnext(address _connext) external onlyOwner {
+    connext = IConnext(_connext);
+    emit SetConnext(_connext);
+  }
+
+  // ======== External: Send Claim =========
+
+  /**
+   * @notice Sends a request to claim the fees in the originated domain
+   * @param _domain The domain where to claim the fees
+   * @param _recipient The address of the relayer
+   * @param _amount The amount of fees to claim
+   * @param _transactionIds A group of transaction ids to claim for fee bumps
+   */
+  function send(
+    uint32 _domain,
+    address _recipient,
+    uint256 _amount,
+    bytes32[] calldata _transactionIds
+  ) external onlyConnext {
+    if (_amount == 0 && _transactionIds.length == 0) revert RelayerFeeRouter__send_claimEmpty();
```

Comment:
Why are we sending `amount` ? @LayneHaber 

I feel like we just need to send the transactionIds, they should be mapped to the corresponding amounts that the user deposited on the origin chain